README.md

```diff
@@ -109,11 +78,70 @@ s3://mybucket/aws-instance-scheduler/v1.4.0/instance-scheduler.template
 s3://mybucket/aws-instance-scheduler/v1.4.0/instance-scheduler-remote.template
 ```
 
-## Deploy
+### Deploy
 
 See the [AWS Instance Scheduler Implementation Guide](https://s3.amazonaws.com/solutions-reference/aws-instance-scheduler/latest/instance-scheduler.pdf) for deployment instructions, using the link to the instance-scheduler.template from your bucket, rather than the one for AWS Solutions. Ex. https://mybucket.s3.amazonaws.com/aws-instance-scheduler/v1.4.0.mybuild/instance-scheduler.template
 
-## CDK Documentation
+
+# Testing the Solution
+
+## Running Tests Locally
+to test the solution you will need to install tox
+
+```pip install tox```
+
+Then from the root directory of the solution
+```
+//run all unit tests
+tox
+
+//test just the lambda code
+tox -e lambda
+
+//test just the cdk code
+tox -e cdk
+```
+
+## Automated Testing Pipeline
+AWS Instance Scheduler comes with an automated testing pipeline that can be used to automatically test any changes you
+develop for the solution on your own development fork
+
+#### _Prerequisites - You must have an AWS account and a fork of this repo_
+
+### Step 1 - Connect CodeStar to Your Github Account
+
+For the pipeline to be able to test your changes, you must provide permission for CodeStar to 
+download the code from your development repo
```

Comment:
CodeStar may only need permissions to clone the repo for this pipeline, but the app you install is granted more permissions than that. The linked documentation covers that, but I wouldn't want to mislead someone into thinking that all this does is allow codestar to clone the repo.

Reply:
Good catch. fixing